### PR TITLE
Bug 1788117: fix CPU and RAM editing in Flavor dialog

### PR DESCRIPTION
**Pull request by suomiy**
- fields can be fully edited
- added a unit field for RAM
- fixed design
- added validations

@pcbailey @mareklibra

### CODE REVIEW

**Comment by suomiy** [comment]
design fixed

aaa
aa

File: frontend/packages/kubevirt-plugin/src/components/modals/vm-flavor-modal/vm-flavor-modal.tsx [hunk 1]
```diff
@@ -1,154 +1,195 @@
...
     cancel,
     loadError,
     loaded,
   } = props;
+  const inProgress = props.inProgress || !loaded;
   const vm = asVM(vmLike);
-  const flattenTemplates = _.get(templates, 'data', []) as TemplateKind[];
+  const flattenTemplates = getLoadedData(templates, []);
 
   const vmFlavor = getFlavor(vmLike);
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/modals/vm-flavor-modal/vm-flavor-modal.tsx]: When VM does not have a flavor `getFlavors` will default to `[custom_flavor]` but `getFlavor` will return `null`
So in cases VM is missing a floavor we will get one option 'custom', but `isCustom` will be false.

I think this is a nice work around:
```
const flavors = getFlavors(vmLike, flattenTemplates);
const vmFlavor = getFlavor(vmLike) || flavors[0];
```

Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/components/modals/vm-flavor-modal/vm-flavor-modal.tsx]: +1, that is much better, but using the last flavor due to the sorting

**Pull request merged**